src/workerd/api/kv-instrumentation-test.js

```diff
@@ -24,21 +24,23 @@ export default {
     return (event) => {
       switch (event.event.type) {
         case 'spanOpen':
-          // The span ids will change between tests, but Map preserves insertion order
-          spans.set(event.spanId, { name: event.event.name });
+          // span ids are simple counters for tests, but invocation ID allows us to differentiate them
+          spans.set(event.invocationId + event.event.spanId, {
+            name: event.event.name,
+          });
           break;
         case 'attributes': {
-          let span = spans.get(event.spanId);
+          let span = spans.get(event.invocationId + event.spanId);
```

Comment:
You're performing this operation (`event.invocationId + event.spanId`) multiple times in this block. Might be easier to maintain to do it once before entering the switch.